### PR TITLE
Update directory structure and use AdoptOpenJDK base images

### PR DESCRIPTION
**Pull request by slide**
Updates the directory structure to follow changes that were made in the agent images. Also, updates to use AdoptOpenJDK as base images instead of OpenJDK. Supersedes #956

### CODE REVIEW

File: 8/debian/buster-slim/hotspot/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,84 @@
+FROM adoptopenjdk/openjdk8:debian-slim
+
+# Install git lfs on Debian stretch per https://github.com/git-lfs/git-lfs/wiki/Installation#debian-and-ubuntu
+# Avoid JENKINS-59569 - git LFS 2.7.1 fails clone with reference repository
+RUN apt-get update && apt-get upgrade -y && apt-get install -y git curl gpg unzip libfreetype6 libfontconfig1 && curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | bash && apt-get install -y git-lfs && git lfs install && rm -rf /var/lib/apt/lists/*
```
Comment (MarkEWaite) [inline 8/debian/buster-slim/hotspot/Dockerfile:5]: Debian Buster does not need the special case install of git LFS.  Should also not need us to specify libfreetype or libfontconfig
```suggestion
# Install git lfs on Debian stretch per https://github.com/git-lfs/git-lfs/wiki/Installation#debian-and-ubuntu
# Avoid JENKINS-59569 - git LFS 2.7.1 fails clone with reference repository
RUN apt-get update && apt-get upgrade -y && apt-get install -y git git-lfs curl gpg unzip && git lfs install && rm -rf /var/lib/apt/lists/*
```

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
We can simplify the install and use the git-lfs provided with the operating system.  It is a new enough version that it does not have the issue we found with Debian stretch

File: 8/debian/buster-slim/hotspot/Dockerfile (hunk 1 above)
Comment (slide) [reply 8/debian/buster-slim/hotspot/Dockerfile:5]: Are you sure on the libfreetype? I think those are needed for the AWT stuff even in headless mode.

Comment (MarkEWaite) [reply 8/debian/buster-slim/hotspot/Dockerfile:5]: Good point.  No objection if we retain libfreetype.  I was able to run the image without it, but may be safer to retain it than risk it.

File: 11/debian/buster-slim/hotspot/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,82 @@
+FROM adoptopenjdk/openjdk11:debian-slim
```
Comment (james-crowley) [inline 11/debian/buster-slim/hotspot/Dockerfile:1]: As we discussed on Gitter, you probably want to pull from `adoptopenjdk/openjdk8:debianslim-slim` as that is made from the Debian Slim builds. The general `-slim` tag you see is mostly a script AdoptOpenJDK runs to remove unneeded files to reduce the overall image size.

**Review by james-crowley — CHANGES_REQUESTED** [review]

**Comment by slide** [comment]
@james-crowley I believe I am using the correct images now, can you look?

File: 11/debian/buster/hotspot/Dockerfile [hunk 3]
```diff
@@ -1,6 +1,6 @@
-FROM openjdk:11-jdk-stretch
+FROM adoptopenjdk:11-hotspot
```
Comment (james-crowley) [inline 11/debian/buster/hotspot/Dockerfile:1]: For official images, it looks like the "default" image is based on Ubuntu not Debian. If your okay with that, you should change the file path to `11/ubuntu/bionic/hotspot/Dockerfile`.

File: 8/debian/buster/hotspot/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-FROM openjdk:8-jdk-slim
+FROM adoptopenjdk:8-hotspot
```
Comment (james-crowley) [inline 8/debian/buster/hotspot/Dockerfile:1]: For official images, it looks like the "default" image is based on Ubuntu not Debian. If your okay with that, you should change the file path to `8/ubuntu/bionic/hotspot/Dockerfile`.

**Review by james-crowley — CHANGES_REQUESTED** [review]
Here is more information: https://github.com/AdoptOpenJDK/openjdk-docker#official-and-unofficial-images

File: 11/debian/buster/hotspot/Dockerfile (hunk 3 above)
Comment (slide) [reply 11/debian/buster/hotspot/Dockerfile:1]: I think we want to stick with debian, so I'll need to update this.

Comment (james-crowley) [reply 11/debian/buster/hotspot/Dockerfile:1]: So I think you would want to use `adoptopenjdk/openjdk11:debian`.

File: 8/debian/buster/hotspot/Dockerfile (hunk 4 above)
Comment (james-crowley) [reply 8/debian/buster/hotspot/Dockerfile:1]: And for this one use `adoptopenjdk/openjdk8:debian`.

**Comment by slide** [comment]
@MarkEWaite have I addressed your concerns?

**Review by MarkEWaite — APPROVED** [review]
Looks good to me.

**Comment by timja** [comment]
publish.sh changes were missed here breaking the docker publishing build on trusted

**Comment by slide** [comment]
I'll fix it up, thanks for the info.

**Pull request merged**